### PR TITLE
HDDS-7473. Ratis integration for support of remove registeration

### PR DESCRIPTION
**Pull request by sumitagrawl**
## What changes were proposed in this pull request?

Integration with ratis for additional cleanup of registeration after ratis provided feature

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-7473

## How was this patch tested?

Integration test is verified for impact. And with debug, its verified for cleanup using debug in ratis.

### CODE REVIEW

**Comment by sumitagrawl** [comment]
@szetszwo Plz review

**Review by szetszwo — COMMENTED** [review]
@sumitagrawl , thanks a lot for working on this!   Please see the comments inlined and also https://issues.apache.org/jira/secure/attachment/13054010/4112_review.patch

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java [hunk 1]
```diff
@@ -114,6 +117,8 @@ public class HddsDatanodeService extends GenericCli implements ServicePlugin {
       new DNMXBeanImpl(HddsVersionInfo.HDDS_VERSION_INFO) { };
   private ObjectName dnInfoBeanName;
   private DatanodeCRLStore dnCRLStore;
+  private List<Pair<Consumer<RatisMetricRegistry>,
+      Consumer<RatisMetricRegistry>>> ratisRegistryList = null;
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java]: Updated...

File: hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java [hunk 2]
```diff
@@ -43,21 +47,34 @@ public RatisDropwizardExports(MetricRegistry registry) {
     super(registry, new RatisNameRewriteSampleBuilder());
   }
 
-  public static void registerRatisMetricReporters(
+  public static List<Pair<Consumer<RatisMetricRegistry>,
+      Consumer<RatisMetricRegistry>>> registerRatisMetricReporters(
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/RatisDropwizardExports.java]: Updated...

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java [hunk 3]
```diff
@@ -284,7 +286,9 @@ public final class StorageContainerManager extends ServiceRuntimeInfoImpl
   // container replicas.
   private ContainerReplicaPendingOps containerReplicaPendingOps;
   private final AtomicBoolean isStopped = new AtomicBoolean(false);
-  
+  private List<Pair<Consumer<RatisMetricRegistry>,
+      Consumer<RatisMetricRegistry>>> ratisRegistryList = null;
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java]: updated ...

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 4]
```diff
@@ -417,6 +420,8 @@ public final class OzoneManager extends ServiceRuntimeInfoImpl
 
   private final OzoneLockProvider ozoneLockProvider;
   private OMPerformanceMetrics perfMetrics;
+  private List<Pair<Consumer<RatisMetricRegistry>,
+      Consumer<RatisMetricRegistry>>> ratisRegistryList = null;
```
[the remark this answers is not in the document]
Comment (sumitagrawl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java]: @szetszwo  all comments updated....

**Pull request merged**